### PR TITLE
templates testing

### PR DESCRIPTION
**Pull request by Ilya-Krylov**
* https://github.com/openvinotoolkit/training_extensions/pull/348 has to be merged first
* added global test that instantiate all templates, checks whether mandatory files have been created, run training, evaluation and export tests

### CODE REVIEW

File: pytorch_toolkit/object_detection/init_venv.sh [hunk 1]
```diff
@@ -2,28 +2,33 @@
...
+  echo "$ . ${venv_dir}/bin/activate"
   exit
 fi
 
 # Download mmdetection
 git submodule update --init ../../external/mmdetection
 
 # Create virtual environment
-virtualenv venv -p python3.6 --prompt="(detection)"
+virtualenv ${venv_dir} -p python --prompt="(detection)"
```
Comment (druzhkov-paul) [inline pytorch_toolkit/object_detection/init_venv.sh:23]: Maybe `python3` or `python` is just ok?

File: pytorch_toolkit/object_detection/tests/common/test_case.py [hunk 2]
```diff
@@ -42,47 +43,98 @@ class TestCaseOteApi(unittest.TestCase):
...
 
-            with open(f'tests/expected_outputs/{problem_name}/{model_name}.json') as read_file:
+            with open(f'{os.path.dirname(__file__)}/../expected_outputs/{problem_name}/{model_name}.json') as read_file:
                 content = json.load(read_file)
 
             self.assertLess(abs(content['map'] - ap / 100), 1e-6)
 
         def test_finetuning(self):
-            log_file = os.path.join(self.work_dir, 'test_finetuning.log')
+                    # def test_evaluation(self):
```
Comment (druzhkov-paul) [inline pytorch_toolkit/object_detection/tests/common/test_case.py:86]: ?

File: pytorch_toolkit/run_test.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import yaml
+
+class GlobalTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        logging.info('Instantiating templates has been started.')
+        venv_dir = tempfile.mkdtemp()
+        cls.work_dir = tempfile.mkdtemp()
+        #cls.work_dir = '/tmp/TEMPO'
```
Comment (druzhkov-paul) [inline pytorch_toolkit/run_test.py:18]:
```suggestion
```

File: pytorch_toolkit/run_test.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import glob
+import os
+from subprocess import run
+import sys
+import tempfile
+import unittest
+import logging
+
+import yaml
```
Comment (druzhkov-paul) [inline pytorch_toolkit/run_test.py:9]: Consider sorting this maybe.

File: pytorch_toolkit/object_detection/init_venv.sh (hunk 1 above)
Comment (Ilya-Krylov) [reply pytorch_toolkit/object_detection/init_venv.sh]: fixed

File: pytorch_toolkit/object_detection/tests/common/test_case.py (hunk 2 above)
Comment (Ilya-Krylov) [reply pytorch_toolkit/object_detection/tests/common/test_case.py]: fixed

File: pytorch_toolkit/run_test.py (hunk 4 above)
Comment (Ilya-Krylov) [reply pytorch_toolkit/run_test.py]: fixed

File: pytorch_toolkit/object_detection/tests/common/test_case.py (hunk 2 above)
Comment (druzhkov-paul) [reply pytorch_toolkit/object_detection/tests/common/test_case.py]: Are u sure?

Comment (Ilya-Krylov) [reply pytorch_toolkit/object_detection/tests/common/test_case.py]: sry, fixed one more time

**Review by druzhkov-paul — APPROVED** [review]

File: pytorch_toolkit/ote/setup.py [hunk 5]
```diff
@@ -1,7 +1,12 @@
+import os
 from setuptools import setup, find_packages
 
+with open(os.path.join(os.path.dirname(__file__), 'requirements.txt')) as read_file:
+    requirements = [requirement for requirement in read_file]
```
Comment (druzhkov-paul) [inline pytorch_toolkit/ote/setup.py:5]: 1.
    ```python
    requirements = list(read_file)
    ```
1. Is it Ok not to strip the lines?

Comment (Ilya-Krylov) [reply pytorch_toolkit/ote/setup.py]: 1. thanks
2. it works actually, but not documented, so .strip() added

**Comment by Ilya-Krylov** [comment]
run tests

**Comment by Ilya-Krylov** [comment]
@DmitriySidnev  @evgeny-izutov  please have a look at `pytorch_toolkit/tests/run_model_templates_tests.py`

**Review by druzhkov-paul — APPROVED** [review]

**Pull request merged**